### ISSUE TITLE
The BatchNorm affine problem in SPOS

### ISSUE THREAD

**Issue by CuriousCat-7** [microsoft/nni#2883]
According to [this](https://github.com/megvii-model/SinglePathOneShot/issues/3#issuecomment-557450100) issue, the BatchNorm's affine parameter should be set to `False` in the training phase and according to the [official retrain code](https://github.com/megvii-model/SinglePathOneShot/blob/master/src/Evaluation/template/network.py), should be set back to `True` in the **retrain phase**. 

But NNI [still use the SPOSOneShot](https://github.com/microsoft/nni/blob/0a21a901227d156cf65ce6330d5fcfc0205c1ffa/examples/nas/spos/scratch.py#L99), which means BatchNorm is set `affine == False`.

I think that's why the NNI [can not reproduce the official result but very close to it](https://nni.readthedocs.io/zh/latest/NAS/SPOS.html#id9).

**Comment by ultmaster** [comment]
Could you share you reproduction result? If your modification improves the reproduction result, please raise a pull request.

Thanks very much.

**Comment by CuriousCat-7** [comment]
> Could you share you reproduction result? If your modification improves the reproduction result, please raise a pull request.
> 
> Thanks very much.

Ok, I will try, with my eight TITAN Xp cards......

**Comment by CuriousCat-7** [comment]
> Could you share you reproduction result? If your modification improves the reproduction result, please raise a pull request.
> 
> Thanks very much.

Could you tell me your GPU configuration? As far as I know, different kinds of GPUs and different numbers of GPUs will get different results in DL training tasks.  I am using 8 TITAN XP cards with CUDA 10.1, and I think your device configuration may be 4 V100 cards, am I right?

**Comment by CuriousCat-7** [comment]
I have got: 
`top1 = 0.728800, top5 = 0.911251`
where "72.88%" is better than the NNI official "72.14%" but still lower than Megvii official "73.61%".
I think the accuracy can still raise if I increase the epoch num, but I don't what to cheat QAQ.

Thanks

**Comment by ultmaster** [comment]
Good news that you've improved the accuracy by ~0.7%.

You are right that my experiments are done with 4 v100 GPUs. Sadly I don't have the log right now.

240 epochs is just a number for fair comparison. I'm sure you can get a better number with, e.g., 1200 epochs, or whatever tricks like learning rate warmup, or cosine learning rate decay.

**Comment by CuriousCat-7** [comment]
Close it cause the issue have been merged. I will try to train it on V100 GPUs and update the result if I succeed at applying the v100 cloud server. Thanks.
